### PR TITLE
default to SHA1 for native

### PR DESCRIPTION
**Pull request by dcousens**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+function nativePBKDF2 (password, salt, iterations, keylen, digest, callback) {
+  return native.pbkdf2(password, salt, iterations, keylen, digest || 'sha1', callback)
+}
+
+function nativePBKDF2Sync (password, salt, iterations, keylen, digest) {
+  return native.pbkdf2Sync(password, salt, iterations, keylen, digest || 'sha1')
+}
+
 /* istanbul ignore next */
-if (crypto && (!crypto.pbkdf2Sync || crypto.pbkdf2Sync.toString().indexOf('keylen, digest') === -1)) {
```
Comment (dcousens) [inline index.js]: `crypto` was removed here,  if it is false, then it would still explode below

File: index.js [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
-  exports.pbkdf2 = require('./lib/async')
-  exports.pbkdf2Sync = require('./lib/sync')
+if (!native.pbkdf2Sync || native.pbkdf2Sync.toString().indexOf('keylen, digest') === -1) {
+  module.exports.pbkdf2Sync = require('./lib/sync')
```
Comment (calvinmetcalf) [inline index.js]: why did you add module. here ?

File: index.js (hunk 1 above)
Comment (calvinmetcalf) [reply index.js]: the issue is if it's undefined not if it's false

Comment (dcousens) [reply index.js]: `false/falsy/undefined`,  the point was the same,   it would explode with the same exception below

File: index.js (hunk 2 above)
Comment (dcousens) [reply index.js]: habit of syntax highlighting,  will remove if needed

Comment (calvinmetcalf) [reply index.js]: lets not add unneded churn

File: index.js (hunk 1 above)
Comment (calvinmetcalf) [reply index.js]: `!native.pbkdf2Sync` will explode *here* if native is `undefined`,  so I guess I may be misunderstanding your fix

Comment (dcousens) [reply index.js]: The removal of `crypto &&` isn't a fix,  its removal of a pointless conditional.

Comment (calvinmetcalf) [reply index.js]: maybe the fix is for it to just give a useful error message if native is undefined instead of blowing up

Comment (dcousens) [reply index.js]: @calvinmetcalf out of interest,  is it a real error condition?
If so,  when?  Lets test for it :+1:

Comment (calvinmetcalf) [reply index.js]: if somebody was to require this directly, but prevent crypto from being loaded then crypto would be undefined here, now *why* I have no idea, probably in some weird enviroment

Comment (dcousens) [reply index.js]: @calvinmetcalf I don't think we need to support that... if it breaks,  we'll learn,  and patch...

**Comment by dcousens** [comment]
Removed unneeded churn,  good to go @calvinmetcalf ?

**Comment by dcousens** [comment]
Another alternative is to drop the `default` and bump the major when the LTS period ends (very soon AFAIK?)

File: index.js (hunk 1 above)
Comment (calvinmetcalf) [reply index.js]: ok I'm still on -1 on this change that could potentially introduce new bugs for purely aesthetic reasons (as far as I can tell)

Comment (dcousens) [reply index.js]: OK,  I'll fix this up

**Comment by calvinmetcalf** [comment]
thanks

**Comment by dcousens** [comment]
Rebased and should be good to go... waiting for tests pass.

**Comment by dcousens** [comment]
@calvinmetcalf 1416637 - `tape` doesn't run synchronously

**Comment by dcousens** [comment]
Tests passing

**Comment by dcousens** [comment]
Shall I merge?

**Comment by calvinmetcalf** [comment]
go for it

**Comment by dcousens** [comment]
Published as `v3.0.16`
